smmregrid/gridtype.py

```diff
@@ -0,0 +1,214 @@
+
+        Raises:
+            ValueError: If more than one vertical dimension is identified.
+        """
+        identified_dims = list(set(self.dims).intersection(default_dims[axis]))
+        if axis == 'vertical':
+            if len(identified_dims) > 1:
+                raise ValueError(f'Only one vertical dimension can be processed at the time: check {identified_dims}')
+            if len(identified_dims) == 1:
+                identified_dims = identified_dims[0]  # unlist the single vertical dimension
+        return identified_dims if identified_dims else None
+
+    def _identify_other_dims(self):
+        """
+        Calculate and return the dimensions that are not part of horizontal_dims, vertical_dim, or time_dims.
+
+        Returns:
+            set: A set of unused dimensions that are not part of horizontal, vertical, or time dimensions.
+        """
+        # Safely convert identified dimensions to sets, handling None values
+        horizontal_dims = set(self.horizontal_dims) if self.horizontal_dims else set()
+        vertical_dim = {self.vertical_dim} if self.vertical_dim else set()  # Use {} for single value
+        time_dims = set(self.time_dims) if self.time_dims else set()
+
+        # Return the unused dimensions by subtracting used dimensions from all dimensions
+        return set(self.dims) - (horizontal_dims | vertical_dim | time_dims)
+
+    def _identify_spatial_bounds(self, data):
+        """
+        Identify bounds variables in the dataset by checking variable names.
+
+        Args:
+            data (xr.Dataset): An xarray dataset containing data variables.
+
+        Returns:
+            list: A list of bounds variable names identified in the dataset.
+        """
+        bounds_variables = []
+
+        for var in data.data_vars:
+            if (var.endswith('_bnds') or var.endswith('_bounds') or var == 'vertices') and 'time' not in var:
+                bounds_variables.append(var)
+
+        return bounds_variables
+
+    # def identify_sizes(self, data):
+    #    """
+    #    Identify the sizes of the dataset based on the horizontal dimensions.
+    #    """
+    #
+    #    if self.horizontal_dims:
+    #        self.horizontal_sizes  = [data.sizes[x] for x in self.horizontal_dims]
+
+    def _identify_variable(self, var_data, var_name=None):
+        """Helper function to process individual variables.
+
+        Args:
+            var_data (xr.DataArray): An xarray DataArray containing variable data.
+            var_name (str, optional): The name of the variable. If None, uses the name from var_data.
+
+        Updates:
+            self.variables: Updates the variables dictionary with the variable's coordinates.
+        """
+        if set(var_data.dims) == set(self.dims):
+            self.variables[var_name or var_data.name] = {
+                'coords': list(var_data.coords),
+            }
+
+    def identify_variables(self, data):
```

Comment:
This function has no "_" in front, suggesting that it could be used from outside. Actually though it only updates `GridType.variables`... so it is more lake an internal method and be prefixed with _ , no?